src/libs/actions/Report.js

```diff
@@ -1084,6 +1086,39 @@ function deleteReportComment(reportID, reportAction) {
                 };
 
                 Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, reportActionsToMerge);
+            } else {
+                // Update last message information in the report object
+                Onyx.connect({
+                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+                    callback: _.once((reportActionList) => {
+                        // Find last non-empty message
+                        const lastReportAction = !_.isEmpty(reportActionList)
```

Comment:
Here the `reportActionList` is an object of objects (as they are stored in local storage), but on line 150 `reportActionList` is an array of objects. Doing `[...reportActionList]` on an object of objects causes an error `<Object> is not iterable`. 

There is also a slight difference in how we get the `message.html` there:
on line 150 it's as simple as `reportAction.message.html`, but here we have to do `_.last(reportActionItem.message).html` because `message` is an array of objects in local storage and not an object with message like it is in API response on line 150.

This was enough for me to ditch the idea of trying to re-use the logic.

Reply:
> _.last(reportActionItem.message).html because message is an array of objects in local storage and not an object with message like it is in API response on line 150.

Ah... that's frustrating, but your logic makes sense then 👍 . I wonder in what scenario we have multiple objects in the `message` array, other than the action being `CREATED` where it looks like we have two...

You can get rid of the trailing comma after `'CREATED'` by doing the following: 

``` Javascript
const lastReportAction = !_.isEmpty(reportActionList)
    ? _.find(Object.values(reportActionList).reverse(),
        reportActionItem => _.last(reportActionItem.message).html !== ''
        && reportActionItem.actionName !== 'CREATED')
    : null;
```

Reply:
I initially thought the edited messages hold the history of edits in that array, but that is not the case.

I've applied the suggested change. Thanks.